src/types.ts

```diff
@@ -385,12 +389,66 @@ export interface OpenSeaAsset extends Asset {
   transferFeePaymentToken: OpenSeaFungibleToken | null
 }
 
+/**
+ * Defines a LastSale type which contains details about last sale of the token
+ */
+export interface LastSale {
+
+  // The type of event
+  eventType: string
+
+  // The timestamp of the transaction (if on-chain) or when the off-chain occurred
+  eventTimestamp: Date
+
+  // The auction type
+  auctionType: string
+
+  // The total price of the sale in the payment
+  totalPrice: string
+
+  // The transaction associated with the token sale
+  transaction?: Transaction
```

Comment:
Not sure if we want to do `transaction: Transaction | null` here instead?

Reply:
yeah let's do that and make it nullable

Reply:
What does it mean when transaction is null ? Does it mean it's pending ?

Reply:
@taylorjdawson @alexanderatallah @diklou 
>What does it mean when transaction is null ? Does it mean it's pending ?

up ⬆️